### PR TITLE
Various cleanups

### PR DESCRIPTION
**Pull request by listx**
This addresses mainly style nits and other smaller things that were uncovered while working on the git client and moonraker. Please have a look at each of the individual commit messages for details.

/cc @cjwagner @airbornepony @timwangmusic

### CODE REVIEW

File: prow/config/secret/reloader.go
Comment (airbornepony) [inline prow/config/secret/reloader.go:60]: Would the message "Unable to stat secret file." help when debugging? (Maybe the err already makes clear it was a failed stat op?)

**Review by airbornepony — APPROVED** [review]

File: prow/config/secret/reloader.go
Comment (listx) [reply prow/config/secret/reloader.go:60]: I don't think it would make much of a difference, because the developer seeing the message would just grep the codebase anyway (regardless of what it says).

**Comment by petr-muller** [comment]
LGTM (not tagging because this still has a hold, feel free to tag me if ready).

I appreciate taking the effort to write motivation commit messages for each change!

**Review by timwangmusic — APPROVED** [review]

**Comment by listx** [comment]
There have been lots of smaller iterative improvements going in (hence why I've been updating this PR a lot). Sorry about the churn!

I'm going to rebase one more time. Marking as WIP for now while the dust settles.

**Comment by listx** [comment]
One of the commits here is in https://github.com/kubernetes/test-infra/pull/30520. When that merges I'll rebase.

**Comment by listx** [comment]
This is blocked by https://github.com/kubernetes/test-infra/pull/30541 (which is in turn blocked by https://github.com/kubernetes/test-infra/pull/30294).

**Comment by listx** [comment]
This is still blocked by #30541, but is otherwise ready for reviews.

File: prow/config/inrepoconfig.go
Comment (timwangmusic) [inline prow/config/inrepoconfig.go:63]: Why is this removed?

Comment (listx) [reply prow/config/inrepoconfig.go:63]: It was renamed. See https://github.com/kubernetes/test-infra/pull/30428/commits/6b1f69dac3ad9c4448d023c13928c0161be2f6c7.

As a rule of thumb, if there are multiple commits in a PR, you should review each commit in isolation. Then you'll get to see which changes belong together (which is not apparent in the combined diff).

**Review by timwangmusic — APPROVED** [review]
LGTM

**Pull request merged**
